### PR TITLE
Feature 147 refine error handling and usability

### LINKED ISSUE #147: Refining error handling and usability
-Remove "Reload page"-button from error dialog
-Update error dialog instructions
-Add snackbar error alert for network error (no internet connection)

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #147 and also adds Generic Snackbar component

### CODE REVIEW

File: src/components/contexts/connection-handler.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      open={ connectionLost }
+      onClose={ () => setConnectionLost(!connectionLost) }
+      severity="error"
+    >
+      <>
+        <Typography variant="h5">{ strings.errorHandling.formScreen.noConnection }</Typography>
+        <Typography variant="caption">{ strings.errorHandling.formScreen.noConnectionHelper }</Typography>
+      </>
+    </GenericSnackbar>
+  );
```
Comment (EVP1989) [inline src/components/contexts/connection-handler.tsx:26]: Move this to bottom, before component return.

File: src/components/contexts/leave-page-handler.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  /**
+   * Handles wheter this components event listeners should be active
+   */
+  const handleActive = () => {
+    if (active) {
+      window.addEventListener("beforeunload", handleUserAlert);
+    } else {
+      window.removeEventListener("beforeunload", handleUserAlert);
+    }
```
Comment (EVP1989) [inline src/components/contexts/leave-page-handler.tsx:33]: Use ternary operator

File: src/components/generic/generic-snackbar.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  anchorOrigin?: {
+    horizontal: "center" | "left" | "right";
+    vertical: "bottom" | "top";
+  };
+  children?: JSX.Element;
+  disableClickaway?: boolean;
+}
+
+/**
+ * GenericSnackbar component
```
Comment (EVP1989) [inline src/components/generic/generic-snackbar.tsx:23]:
```suggestion
 * Generic snackbar component
```

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/components/contexts/connection-handler.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/contexts/connection-handler.tsx:26]: Done

File: src/components/contexts/leave-page-handler.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/contexts/leave-page-handler.tsx:33]: Done

File: src/components/generic/generic-snackbar.tsx (hunk 3 above)
Comment (villejuutila) [reply src/components/generic/generic-snackbar.tsx:23]: Done

File: src/components/contexts/connection-handler.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { Typography } from "@mui/material";
+import GenericSnackbar from "components/generic/generic-snackbar";
+import strings from "localization/strings";
+import React, { useEffect, useState } from "react";
+
+/**
+ * ConnectionHandler component
+ */
+const ConnectionHandler: React.FC = ({ children }) => {
```
Comment (EVP1989) [inline src/components/contexts/connection-handler.tsx:9]: Move FC to React import

File: src/components/contexts/leave-page-handler.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+interface Props {
+  active: boolean;
+  children: JSX.Element;
+}
+
+/**
+ * LeavePageHandler component
+ */
+const LeavePageHandler: React.FC<Props> = ({
```
Comment (EVP1989) [inline src/components/contexts/leave-page-handler.tsx:11]: Move FC to React import

File: src/components/generic/generic-snackbar.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    vertical: "bottom" | "top";
+  };
+  children?: JSX.Element;
+  disableClickaway?: boolean;
+}
+
+/**
+ * Generic Snackbar component
+ */
+const GenericSnackbar: React.FC<Props> = ({
```
Comment (EVP1989) [inline src/components/generic/generic-snackbar.tsx:25]: Move FC to React import

**Review by EVP1989 — APPROVED** [review]
Approved with comments

File: src/components/contexts/connection-handler.tsx (hunk 4 above)
Comment (villejuutila) [reply src/components/contexts/connection-handler.tsx]: Done

File: src/components/generic/generic-snackbar.tsx (hunk 6 above)
Comment (villejuutila) [reply src/components/generic/generic-snackbar.tsx]: Done

File: src/components/contexts/leave-page-handler.tsx (hunk 5 above)
Comment (villejuutila) [reply src/components/contexts/leave-page-handler.tsx]: Done

**Pull request merged**